IntegrationWithPharmacies/Controllers/SharingPrescriptionController.cs

```diff
@@ -1,30 +1,198 @@
+            medicines.Add(medicine);
+        }
+
+        private int getRandomNumber()
+        {
+            return new Random().Next(1, 100);
+        }
+        private String getTextForPrescription(Prescription prescription)
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            return stringBuilder.Append(prescription.Pharmacy+"          Precription for medicine\n\nPatients name: " + prescription.Name + "\nPatients surname: " + prescription.Surname + "\nPatients medical ID number: " + prescription.MedicalIDNumber + "\nMedication: " + prescription.Medicine + "     Quantity: " + prescription.Quantity + "\nUsage: " + prescription.Usage + "\n").ToString();
+         
+        }
+        [HttpPost("http")]
+        public IActionResult PostHttp(Prescription prescription)
+        {
+            if (Environment.Equals("Development"))return sendPrescriptionHttp(prescription);
+            return BadRequest();
+        }
+
+        private IActionResult sendPrescriptionHttp(Prescription prescription)
+        {
+            String complete = @"FilePrescriptions\..\Prescription" + DateTime.Now.ToString("dd-MM-yyyy") + "_" + getRandomNumber() + ".txt";
+            System.IO.FileStream fs = System.IO.File.Create(complete);
+            fs.Close();
+            System.IO.File.WriteAllText(complete, getTextForPrescription(prescription));
+            try
+            {   uploadFile(complete);
+                return Ok();
+            }
+            catch (Exception e) { return BadRequest(); }
+           
+        }
+
+        public void uploadFile(String complete)
+        {
+            WebClient client = new WebClient();
+            Uri uri = new Uri(@"http://localhost:8082/download/prescription/http");
+            client.Credentials = CredentialCache.DefaultCredentials;
+            client.UploadFile(uri, "POST", complete);
+            client.Dispose();
+            SendNotificationAboutReport();
+        }
+
+        public void SendNotificationAboutReport()
+        {
+            try { sendEmail();  }
+            catch (SmtpException ex) { Console.WriteLine(ex.Message); }
+        }
+
+        private static void sendEmail()
+        {   SmtpClient SmptServer = new SmtpClient("smtp.gmail.com");
+            SmptServer.Port = 587;
+            SmptServer.Credentials = new System.Net.NetworkCredential("ourhospital9@gmail.com", "hospital.9");
+            SmptServer.EnableSsl = true;
+            SmptServer.Send(GetMailInformation());
+        }
+
+        private static MailMessage GetMailInformation()
+        {
+            MailMessage mail = new MailMessage();
+            mail.From = new MailAddress("ourhospital9@gmail.com");
+            mail.To.Add("pharmacyisa@gmail.com");
+            mail.Subject = "Notification about sent file";
+            mail.Body = "Body of mail address";
+            return mail;

```

Comment:
Create a constructor (if it doesn't exist) according to the needs of the method, and call it.